cni/pkg/ambient/informers.go

```diff
@@ -149,7 +149,7 @@ func (s *Server) handleUpdate(event controllers.Event) error {
 	notAmbientPod := !wasEnabled && !nowEnabled
 	removeFromMesh := wasEnabled && !nowEnabled
 	joinMesh := !wasEnabled && nowEnabled
-	meshedPodNotInIpset := nowEnabled && !IsPodInIpset(newPod)
```

Comment:
Reverted to [the original logic](https://github.com/istio/istio/pull/47843/files#diff-f294b0dbffa3bed81476df25b8827a991a31c5f21a8fe19b8351adbfd66a7281L138)